src/main/index.js

```diff
@@ -152,6 +158,85 @@ function runApp() {
       })
     })
 
+    protocol.registerBufferProtocol('imagecache', (request, callback) => {
+      const url = decodeURIComponent(request.url.substring(13))
+      if (imageCache.has(url)) {
+        const cached = imageCache.get(url)
+
+        // eslint-disable-next-line node/no-callback-literal
+        callback({
+          mimeType: cached.mimeType,
+          data: cached.data
+        })
+      } else {
+        const newRequest = net.request({
+          method: request.method,
+          url
+        })
+
+        // Electron doesn't allow certain headers to be set:
+        // https://www.electronjs.org/docs/latest/api/client-request#requestsetheadername-value
+        // also blacklist Origin and Referrer as we don't want to let YouTube know about them
+        const blacklistedHeaders = ['content-length', 'host', 'trailer', 'te', 'upgrade', 'cookie2', 'keep-alive', 'transfer-encoding', 'origin', 'referrer']
+
+        for (const header of Object.keys(request.headers)) {
+          if (!blacklistedHeaders.includes(header.toLowerCase())) {
+            newRequest.setHeader(header, request.headers[header])
+          }
+        }
+
+        newRequest.on('response', (response) => {
+          const chunks = []
+          response.on('data', (chunk) => {
+            chunks.push(chunk)
+          })
+
+          response.on('end', () => {
+            const data = Buffer.concat(chunks)
+
+            const age = parseInt(response.headers.age ?? 0)
+            const maxAge = parseInt(response.headers['cache-control'].match(/max-age=([0-9]+)/)[1])
```

Comment:
I've added fallbacks now.
1. It will try to use the max-age in the cache-control header if it is available, alternatively it will use the expires header, otherwise it will fallback to a hardcoded expiry time (currently 2 hours).